browser/js/common/factories/JobApplication.js

```diff
@@ -1,18 +1,122 @@
-
 app.factory('JobApplication', function($http) {
-    var JobApplication = {};
+  var JobApplication = {};
+
+  let fields = [{
+    noFormControl: true,
+    className: 'col-md-12',
+    template: '<h4 class="section-title">PERSONAL INFO</h4>'
+  }];
 
-    JobApplication.create = function (jobApp) {
-        return $http.post('/api/jobs/applications', jobApp)
-        .then(function (res) {
-            return res.data;
+  function generateLinks(links) {
+    for (let link in links) {
+      let inputConfig = links[link].value;
+      if (inputConfig === 0 || inputConfig === 1) {
+        fields.push({
+          noFormControl: true,
+          className: 'col-md-3',
+          template: '<h5 class="default-input-labels">' + links[link].label + '</h5>'
+        })
+        fields.push({
+          key: link,
+          type: 'input',
+          className: 'col-md-8 default-inputs',
+          templateOptions: {
+            type: 'text',
+            required: (inputConfig === 0) ? true : false
+          }
         })
+      }
     }
+    fields.push({
+      noFormControl: true,
+      className: 'col-md-12',
+      template: '<h4 class="section-title">QUESTIONS</h4>'
+    })
+  }
+
+  function generatePersonalInfo(personalInfo) {
+    for (let field in personalInfo) {
+      let fieldConfig = personalInfo[field].value;
+      if (fieldConfig === 0 || fieldConfig === 1) {
+        fields.push({
```

Comment:
Not crazy about all this statefullness. Instead of pushing to the same closed over `fields` array everywhere, maybe just create a new array, push to that and return it. Then later on, concatenate all the arrays together?